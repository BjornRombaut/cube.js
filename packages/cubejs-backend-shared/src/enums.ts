/**
 * @license Apache-2.0
 * @copyright Cube Dev, Inc.
 * @fileoverview
 * Shared backend enums and constants.
 */

/**
 * Queries aliases strings.
 */
export enum QueryAlias {
  TOTAL_COUNT = 'total_count',
  ORIGINAL_QUERY = 'original_query',
  LAMBDA_RANGE_START = 'start',
  LAMBDA_RANGE_END = 'end',
}
