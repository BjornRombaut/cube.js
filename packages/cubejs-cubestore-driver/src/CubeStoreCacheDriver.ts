import { createCancelablePromise, MaybeCancelablePromise } from '@cubejs-backend/shared';
import { CacheDriverInterface } from '@cubejs-backend/base-driver';

import { CubeStoreDriver } from './CubeStoreDriver';

interface CubeStoreCacheDriverOptions {}

export class CubeStoreCacheDriver implements CacheDriverInterface {
  protected readonly connection: CubeStoreDriver;

  public constructor(options: CubeStoreCacheDriverOptions) {
    this.connection = new CubeStoreDriver({});
  }

  // protected async getClient() {
  //   return this.connection.getClient();
  // }

  public async get(key: string) {
    // const client = await this.getClient();
    //
    // try {
    //   const res = await client.getAsync(key);
    //   return res && JSON.parse(res);
    // } finally {
    //   this.redisPool.release(client);
    // }

    throw new Error('Unimplemented');
  }

  public withLock = (
    key: string,
    cb: () => MaybeCancelablePromise<any>,
    expiration: number = 60,
    freeAfter: boolean = true,
  ) => createCancelablePromise(async (tkn) => {
    // const client = await this.getClient();
    //
    // try {
    //   if (tkn.isCanceled()) {
    //     return false;
    //   }
    //
    //   const response = await client.setAsync(
    //     key,
    //     '1',
    //     // Only set the key if it does not already exist.
    //     'NX',
    //     'EX',
    //     expiration
    //   );
    //
    //   if (response === 'OK') {
    //     try {
    //       await tkn.with(cb());
    //     } finally {
    //       if (freeAfter) {
    //         await client.delAsync(key);
    //       }
    //     }
    //
    //     return true;
    //   }
    //
    //   return false;
    // } finally {
    //   this.redisPool.release(client);
    // }

    throw new Error('Unimplemented');
  });

  // @ts-ignore
  public async set(key: string, value, expiration) {
    // const client = await this.getClient();
    //
    // try {
    //   const strValue = JSON.stringify(value);
    //   await client.setAsync(key, strValue, 'EX', expiration);
    //   return {
    //     key,
    //     bytes: Buffer.byteLength(strValue),
    //   };
    // } finally {
    //   this.redisPool.release(client);
    // }

    throw new Error('Unimplemented');
  }

  // @ts-ignore
  public async remove(key: string) {
    // const client = await this.getClient();
    //
    // try {
    //   return await client.delAsync(key);
    // } finally {
    //   this.redisPool.release(client);
    // }

    throw new Error('Unimplemented');
  }

  // @ts-ignore
  public async keysStartingWith(prefix: string) {
    // const client = await this.getClient();
    //
    // try {
    //   return await client.keysAsync(`${prefix}*`);
    // } finally {
    //   this.redisPool.release(client);
    // }

    throw new Error('Unimplemented');
  }

  public async cleanup(): Promise<void> {
    //
  }

  public async testConnection(): Promise<void> {
    // const client = await this.getClient();
    //
    // try {
    //   await client.ping();
    // } finally {
    //   this.redisPool.release(client);
    // }

    throw new Error('Unimplemented');
  }
}
