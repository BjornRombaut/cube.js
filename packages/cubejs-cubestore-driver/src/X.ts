import { CubeStoreDriver } from './CubeStoreDriver';

export class X extends CubeStoreDriver {
  public constructor() {
    super({
      host: '127.0.0.1',
      user: undefined,
      password: undefined,
      port: 3030,
    });
  }
}