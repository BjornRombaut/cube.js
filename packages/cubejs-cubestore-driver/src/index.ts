export * from './CubeStoreQuery';
export * from './CubeStoreDriver';
export * from './CubeStoreDevDriver';
export * from './rexport';
export * from './X';
