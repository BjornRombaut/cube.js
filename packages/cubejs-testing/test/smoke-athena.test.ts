import cubejs, { CubejsApi } from '@cubejs-client/core';
// eslint-disable-next-line import/no-extraneous-dependencies
import { afterAll, beforeAll, expect, jest } from '@jest/globals';
import { BirdBox, getBirdbox } from '../src';
import { DEFAULT_CONFIG, testQueryMeasure } from './smoke-tests';

async function setupBirdbox(extraEnv: Record<string, string> = {}) {
  return getBirdbox(
    'athena',
    {
      CUBEJS_DB_TYPE: 'athena',
      ...DEFAULT_CONFIG,
      ...extraEnv,
    },
    {
      schemaDir: 'postgresql/schema',
    }
  );
}

async function setupClient(birdbox: BirdBox) {
  return cubejs(async () => 'test', {
    apiUrl: birdbox.configuration.apiUrl,
  });
}

async function runScheduledRefresh(client: any) {
  return client.loadMethod(
    () => client.request('run-scheduled-refresh'),
    (response: any) => response,
  );
}

describe('athena', () => {
  jest.setTimeout(60 * 5 * 1000);

  // test('query measure', async () => {
  //   const birdbox = await setupBirdbox();
  //   try {
  //     const client = await setupClient(birdbox);
  //     await testQueryMeasure(client);
  //   } finally {
  //     await birdbox.stop();
  //   }
  // });

  test('rollup measure', async () => {
    const birdbox = await setupBirdbox({
      CUBEJS_REFRESH_WORKER: 'true',
      CUBEJS_ROLLUP_ONLY: 'true',
    });
    try {
      const client = await setupClient(birdbox);
      // await client.runScheduledRefresh();
      await runScheduledRefresh(client);
      await testQueryMeasure(client);
    } finally {
      await birdbox.stop();
    }
  });
});
