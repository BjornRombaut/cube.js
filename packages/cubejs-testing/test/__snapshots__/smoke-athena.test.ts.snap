// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`athena query measure: query 1`] = `
Array [
  Object {
    "Orders.totalAmount": "1700",
  },
]
`;

exports[`athena rollup measure: query 1`] = `
Array [
  Object {
    "Orders.totalAmount": "1700",
  },
]
`;
